app/src/main/java/com/esri/arcgisruntime/opensourceapps/datacollection/util/Event.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2020 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.esri.arcgisruntime.opensourceapps.datacollection.util
+
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+
+/**
+ * Used as a wrapper for data that is exposed via a LiveData that represents an event.
+ *
+ * <p>
+ * This avoids a common problem with events: on configuration change (like rotation) an update
+ * can be emitted if the observer is active. This LiveData only calls the observable if there's an
+ * explicit call to postValue().
+ * <p>
+ */
+open class Event<out T>(private val content: T) {
```

Comment:
I think the class name needs to convene that this event only fires once. How about something like `OneOffEvent` or `SingleEvent`?

Reply:
Does `Event` does not convey that ? I thought of it as `Event` is liveData that only calls the Observable when the Event is raised...

I am leaning towards `SingleEvent` if you really think we need the name change.

Reply:
Ok, I can't think of a better name either. If we ever implement other types of "events", we may have to rename this one, but for now let's keep the name as is.